src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs

```diff
@@ -70,7 +69,14 @@ public static class AdoNetIntegration
             }
             catch (Exception ex)
             {
-                Log.ErrorException($"Error resolving {SystemDataCommonDbCommand}.{nameof(ExecuteDbDataReader)}(...)", ex);
+                Log.ErrorRetrievingMethod(
+                    exception: ex,
+                    moduleVersionPointer: moduleVersionPtr,
+                    mdToken: mdToken,
+                    opCode: opCode,
+                    instrumentedType: DbCommandTypeName,
+                    methodName: nameof(ExecuteDbDataReader),
+                    instanceType: @this?.GetType().AssemblyQualifiedName);
```

Comment:
In other integrations, you added checks for `this == null` and throw `ArgumentNullException`. We should probably make this consistent.

Reply:
I agree, it just felt like a lot more to change for this PR.